codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -833,8 +834,18 @@ public Http2Stream closeRemoteSide() {
      */
     private final class DefaultEndpoint<F extends Http2FlowController> implements Endpoint<F> {
         private final boolean server;
+        /**
+         * This is the next stream id that this endpoint expects to see when creating a new stream.
+         */
         private int nextStreamId;
-        private int lastStreamCreated;
+        /**
+         * Used to generate stream ids. This is different from {@link #nextStreamId} because this may be used to
+         * create stream ids for stream that may not be created immediately. For example if events need to be queued
+         * to respect the
+         * <a href="https://tools.ietf.org/html/rfc7540#section-6.5.2">SETTINGS_MAX_CONCURRENT_STREAMS</a> this can be
+         * used to ensure stream ids continue to increase.
+         */
+        private int nextStreamIdToCreate;
```

Comment:
I think your descriptions should be flipped relative to what the variables are currently used for. The comments you gave here apply to the other variable....however I can flip the names because i think that makes more sense.

`nextStreamToCreate` is fine with me